File: PS/Project1/12014.c
#if 0
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <stdlib.h>

int T, N, arr[10001], len[10001], mlen, tem, ans,K;

int binary(int ml, int find) {
	int left = 1, right = ml, mid;

	while (left < right) {
		mid = (left + right) / 2;

		if (len[mid] > find) {
			right = mid;
		}
		else {
			left = mid + 1;
		}
	}
	return left;
}

int main() {
	scanf("%d", &T);
	for (int tc = 1; tc <= T; tc++) {
		scanf("%d %d", &N, &K);

		for (int i = 1; i <= N; i++) {
			scanf("%d", &arr[i]);
		}
		ans = 0;
		mlen = 0;
		for (int j = 1; j <= N; j++) {
			if (len[mlen] < arr[j]) {
				len[mlen + 1] = arr[j];
				mlen++;
			}
			else {
				tem = binary(mlen, arr[j]);
				len[tem] = arr[j];
			}
			if (mlen == K) {
				ans = 1;
				break;
			}
		}
		printf("Case #%d\n%d\n",tc, ans);
	}
	return 0;
}

#endif

File: Python/12014.py
import sys
ports = [0 for _ in range(10000+1)]
num = [0 for _ in range(10000+1)]

def binary(right, find):
    le,ri = 1, right
    while (le <ri):
        mid = (le+ri)//2
        if num[mid] <= find:
            le = mid+1
        else:
            ri = mid
    return le
T= int(input())
for tc in range(1,T+1):
    N,K = map(int, sys.stdin.readline().split())

    ports= [0]+list(map(int, sys.stdin.readline().split()))
    ans =0
    for i in range(1, N+1):
        maxlen = 0
        for j in range(i, N+1):
            if num[maxlen] < ports[j]:
                num[maxlen+1] = ports[j]
                maxlen+=1
            else:
                tem = binary(maxlen, ports[j])
                num[tem] = ports[j]
            if K==maxlen:
                ans=1
                break
    print("Case #{}".format(tc))
    print(ans)